### PR TITLE
add linworks favepada

### PR DESCRIPTION
**Pull request by duilchoi**
add favepada

## Description
hi. 
i added linworks favepada.
please review .

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
most of the config can be transferred to the info.json file (and should be).

`qmk info -kb linworks/favepada -f json` should help with that

File: keyboards/linworks/favepada/info.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["A14", "A15", "B3", "B4"],
+        "rows": ["A9", "A8", "B15", "B14", "B13", "B12"]
+    },
+    "mouse_key": {
+        "enabled": false
+    },
```
Comment (drashna) [inline keyboards/linworks/favepada/info.json:31]:
```suggestion
```

File: keyboards/linworks/favepada/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        MO(1),    KC_A,    KC_B,    KC_C,
+        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,    KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,    KC_P5,   KC_P6,   
+        KC_P1,    KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,             KC_PDOT 
+    ),
```
Comment (drashna) [inline keyboards/linworks/favepada/keymaps/default/keymap.c:26]: Should be reformatted like this, as this conforms to the numpad_6x4 community layout.

```suggestion
    [0] = LAYOUT_numpad_6x4(
        MO(1),    KC_A,    KC_B,    KC_C,
        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,    KC_P2,   KC_P3,
        KC_P0,             KC_PDOT, KC_PENT
    ),
```

File: keyboards/linworks/favepada/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# linworks/favepada
+
+![linworks/favepada front](https://imgur.com/tkKHcL2)
+![linworks/favepada back](https://imgur.com/1x5UokR)
```
Comment (drashna) [inline keyboards/linworks/favepada/readme.md:4]:
```suggestion
![linworks/favepada front](https://i.imgur.com/tkKHcL2h.png)
![linworks/favepada back](https://i.imgur.com/1x5UokRh.png)
```

File: keyboards/linworks/favepada/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+*A short description of the keyboard/project*
```
Comment (drashna) [inline keyboards/linworks/favepada/readme.md:6]: ?

File: keyboards/linworks/favepada/readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+* Keyboard Maintainer: [keydler](https://github.com/duilchoi)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (drashna) [inline keyboards/linworks/favepada/readme.md:10]: ?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/readme.md (hunk 4 above)
Comment (duilchoi) [reply keyboards/linworks/favepada/readme.md:6]: my mistake!

File: keyboards/linworks/favepada/config.h [hunk 6]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Define RGB */
+#define RGBLED_NUM 37
+#define RGB_MATRIX_LED_COUNT 37
+#define RGB_DI_PIN B1
```
Comment (drashna) [inline keyboards/linworks/favepada/config.h:7]:
```suggestion
#define WS2812_DI_PIN B1
#define RGB_DI_PIN WS2812_DI_PIN
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/info.json [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    "manufacturer": "Lx3",
+    "keyboard_name": "favepada",
+    "maintainer": "keydler",
+    "bootloader": "stm32-dfu",
+    "processor": "STM32F072",
+    "debounce": 5,
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": false
+    },
```
Comment (drashna) [inline keyboards/linworks/favepada/info.json:11]:
```suggestion
```

File: keyboards/linworks/favepada/info.json [hunk 8]
```diff
@@ -0,0 +1,104 @@
+{
+    "manufacturer": "Lx3",
+    "keyboard_name": "favepada",
+    "maintainer": "keydler",
+    "bootloader": "stm32-dfu",
+    "processor": "STM32F072",
+    "debounce": 5,
```
Comment (drashna) [inline keyboards/linworks/favepada/info.json:7]: This is the default value, and doesn't need to be set here.
```suggestion
```

File: keyboards/linworks/favepada/info.json [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    "encoder": {
+        "enabled": false
+    },
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
```
Comment (drashna) [inline keyboards/linworks/favepada/info.json:18]:
```suggestion
        "encoder": true,
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/config.h [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+// The PIXEL_FRACTAL effect does not work properly when the matrix layout is
+// different from the physical layout; it also has problems when underglow
+// LEDs are present, or when multiple LEDs are associated with the same key.
+#undef ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+
+// Framebuffer effects; can be enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS
+// is defined.  Both of these effects currently don't work properly when the
+// key matrix does not match the physical layout, so they are disabled.
+#undef ENABLE_RGB_MATRIX_TYPING_HEATMAP
+#undef ENABLE_RGB_MATRIX_DIGITAL_RAIN
```
Comment (drashna) [inline keyboards/linworks/favepada/config.h:54]: same as other PR.  If they're not enabled, they don't need to be undef'ed

Comment (duilchoi) [reply keyboards/linworks/favepada/config.h:54]: thanks bro!

File: keyboards/linworks/favepada/config.h [hunk 11]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Define RGB */
```
Comment (drashna) [inline keyboards/linworks/favepada/config.h:4]:
```suggestion
#pragma once

/* Define RGB */
```

File: keyboards/linworks/favepada/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    [2] = LAYOUT_numpad_6x4(
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,    KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,    KC_P5,   KC_P6,   
+        KC_P1,    KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,             KC_PDOT
```
Comment (drashna) [inline keyboards/linworks/favepada/keymaps/default/keymap.c:41]:
```suggestion
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
        KC_P1,    KC_P2,   KC_P3,   
        KC_P0,             KC_PDOT, KC_PENT
    ),
    [2] = LAYOUT_numpad_6x4(
        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
        KC_P1,    KC_P2,   KC_P3,   
        KC_P0,             KC_PDOT, KC_PENT 
```

File: keyboards/linworks/favepada/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    ),
+    [2] = LAYOUT_numpad_6x4(
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,    KC_P8,   KC_P9,   KC_PPLS,
+        KC_P4,    KC_P5,   KC_P6,   
+        KC_P1,    KC_P2,   KC_P3,   KC_PENT,
+        KC_P0,             KC_PDOT 
+    )
+};
```
Comment (drashna) [inline keyboards/linworks/favepada/keymaps/via/keymap.c:43]:
```suggestion
    [0] = LAYOUT_numpad_6x4(
        MO(1),    KC_A,    KC_B,    KC_C,
        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
        KC_P1,    KC_P2,   KC_P3, 
        KC_P0,             KC_PDOT, KC_PENT 
    ),
    [1] = LAYOUT_numpad_6x4(
        KC_TRNS,  RGB_TOG, RGB_MOD, KC_D,
        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
        KC_P1,    KC_P2,   KC_P3,   
        KC_P0,             KC_PDOT, KC_PENT
    ),
    [2] = LAYOUT_numpad_6x4(
        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,    KC_P8,   KC_P9,
        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
        KC_P1,    KC_P2,   KC_P3,   
        KC_P0,             KC_PDOT, KC_PENT 
    )
};
```

File: keyboards/linworks/favepada/info.json [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+                { "label": "4", "matrix": [3, 0], "x": 0, "y": 3.25 },
+                { "label": "5", "matrix": [3, 1], "x": 1, "y": 3.25 },
+                { "label": "6", "matrix": [3, 2], "x": 2, "y": 3.25 },
+                { "label": "1", "matrix": [4, 0], "x": 0, "y": 4.25 },
+                { "label": "2", "matrix": [4, 1], "x": 1, "y": 4.25 },
+                { "label": "3", "matrix": [4, 2], "x": 2, "y": 4.25 },
+                { "h": 2, "label": "Enter", "matrix": [4, 3], "x": 3, "y": 4.25 },
+                { "label": "0", "matrix": [5, 0], "w": 2, "x": 0, "y": 5.25 },
+                { "label": "Del", "matrix": [5, 2], "x": 2, "y": 5.25 }
+            ]
```
Comment (drashna) [inline keyboards/linworks/favepada/info.json:97]:
```suggestion
                { "label": "4", "matrix": [3, 0], "x": 0, "y": 3.25 },
                { "label": "5", "matrix": [3, 1], "x": 1, "y": 3.25 },
                { "label": "6", "matrix": [3, 2], "x": 2, "y": 3.25 },
                { "h": 2, "label": "+", "matrix": [2, 3], "x": 3, "y": 2.25 },
                { "label": "1", "matrix": [4, 0], "x": 0, "y": 4.25 },
                { "label": "2", "matrix": [4, 1], "x": 1, "y": 4.25 },
                { "label": "3", "matrix": [4, 2], "x": 2, "y": 4.25 },
                { "label": "0", "matrix": [5, 0], "w": 2, "x": 0, "y": 5.25 },
                { "label": "Del", "matrix": [5, 2], "x": 2, "y": 5.25 },
                { "h": 2, "label": "Enter", "matrix": [4, 3], "x": 3, "y": 4.25 }
            ]
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/rules.mk [hunk 15]
```diff
@@ -0,0 +1,2 @@
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (drashna) [inline keyboards/linworks/favepada/rules.mk:2]: This is set by default on develop.  Can be removed.
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/linworks/favepada/info.json [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+            { "flags": 4, "x": 224, "y": 31 },
+            { "flags": 4, "x": 224, "y": 18 }
+        ]
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x000B",
+        "vid": "0x4C58"
+    },
```
Comment (dunk2k) [inline keyboards/linworks/favepada/info.json:72]:
```suggestion
    },
    "community_layouts": ["numpad_6x4"],
```

File: keyboards/linworks/favepada/config.h
Comment (dunk2k) [inline keyboards/linworks/favepada/config.h:1]: can settings in this file that have data-driven equivalent be migrated to `info.json`.

`qmk migrate`, in qmk cli, should be able to assist with this.

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/config.h
Comment (duilchoi) [reply keyboards/linworks/favepada/config.h:1]: thanks bro!
mission complete!

File: keyboards/linworks/favepada/config.h [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+#endif
```
Comment (dunk2k) [inline keyboards/linworks/favepada/config.h:71]: migrate to `info.json`
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
#define RGB_MATRIX_KEYPRESSES 
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
```

File: keyboards/linworks/favepada/info.json [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+            { "flags": 4, "x": 224, "y": 56 },
+            { "flags": 4, "x": 224, "y": 43 },
+            { "flags": 4, "x": 224, "y": 31 },
+            { "flags": 4, "x": 224, "y": 18 }
+        ],
+        "max_brightness": 120
+    },
+    "rgblight": {
+        "led_count": 37
+    },
```
Comment (dunk2k) [inline keyboards/linworks/favepada/info.json:70]: 1. RGB_MATRIX led count is in `config.h`
2. rgblight is disabled
3. WS2812 pin migrated from `config.h`
```suggestion
    "ws2812": {
        "pin": "B1"
    },
```

File: keyboards/linworks/favepada/info.json [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["A14", "A15", "B3", "B4"],
+        "rows": ["A9", "A8", "B15", "B14", "B13", "B12"]
+    },
+    "processor": "STM32F072",
+    "rgb_matrix": {
```
Comment (dunk2k) [inline keyboards/linworks/favepada/info.json:25]: enable_rgb_matrix_* migrated from `config.h`
```suggestion
    "rgb_matrix": {
        "animations": {
            "alphas_mods": true,
            "gradient_up_down": true,
            "gradient_left_right": true,
            "breathing": true,
            "band_sat": true,
            "band_val": true,
            "band_pinwheel_sat": true,
            "band_pinwheel_val": true,
            "band_spiral_sat": true,
            "band_spiral_val": true,
            "cycle_all": true,
            "cycle_left_right": true,
            "cycle_up_down": true,
            "rainbow_moving_chevron": true,
            "cycle_out_in": true,
            "cycle_out_in_dual": true,
            "cycle_pinwheel": true,
            "cycle_spiral": true,
            "dual_beacon": true,
            "rainbow_beacon": true,
            "rainbow_pinwheels": true,
            "raindrops": true,
            "jellybean_raindrops": true,
            "hue_breathing": true,
            "hue_pendulum": true,
            "hue_wave": true,
            "pixel_rain": true,
             "solid_reactive_simple": true,
            "solid_reactive": true,
            "solid_reactive_wide": true,
            "solid_reactive_multiwide": true,
            "solid_reactive_cross": true,
            "solid_reactive_multicross": true,
            "solid_reactive_nexus": true,
            "solid_reactive_multinexus": true,
            "splash": true,
            "multisplash": true,
            "solid_splash": true,
            "solid_multisplash": true
        },
```

File: keyboards/linworks/favepada/readme.md
Comment (dunk2k) [inline keyboards/linworks/favepada/readme.md:1]: This file is missing bootloader instructions - please see `/data/templates/keyboard/readme.md` for reference.

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/readme.md
Comment (dunk2k) [reply keyboards/linworks/favepada/readme.md:1]: :bump:

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/linworks/favepada/info.json [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (waffle87) [inline keyboards/linworks/favepada/info.json:18]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true,
```

File: keyboards/linworks/favepada/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+        KC_P1,    KC_P2,   KC_P3,   
+        KC_P0,             KC_PDOT, KC_PENT
+    ),
+    [2] = LAYOUT_numpad_6x4(
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,    KC_P8,   KC_P9,
+        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
+        KC_P1,    KC_P2,   KC_P3,   
+        KC_P0,             KC_PDOT, KC_PENT
```
Comment (waffle87) [inline keyboards/linworks/favepada/keymaps/default/keymap.c:41]: This layer is not accessible.
```suggestion
```

File: keyboards/linworks/favepada/keymaps/via/keymap.c [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+        KC_P1,    KC_P2,   KC_P3,   
+        KC_P0,             KC_PDOT, KC_PENT
+    ),
+    [2] = LAYOUT_numpad_6x4(
+        KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_NUM,   KC_PSLS, KC_PAST, KC_PMNS,
+        KC_P7,    KC_P8,   KC_P9,
+        KC_P4,    KC_P5,   KC_P6,   KC_PPLS,   
+        KC_P1,    KC_P2,   KC_P3,   
+        KC_P0,             KC_PDOT, KC_PENT
```
Comment (waffle87) [inline keyboards/linworks/favepada/keymaps/via/keymap.c:41]: This layer is not accessible.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**